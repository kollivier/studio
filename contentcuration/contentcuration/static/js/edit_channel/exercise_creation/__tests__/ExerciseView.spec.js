const Backbone = require('backbone');
const ExerciseView = require('../views').ExerciseView;
// Register handlebars helpers to ensure templates are rendered properly.
require('handlebars/helpers');

const exampleAssessment = {
  type: 'multiplechoice',
  question: "Ceci n'est pas un question?",
  hints: '["Maybe it is, maybe not?"]',
  answers:
    '[{ "correct": true, "answer": "First" }, { "correct": false, "answer": "Second" }, { "correct": false, "answer": "Fish" } ]',
  order: 1,
  assessment_id: 'test',
  raw_data: '',
  source_url: '',
  randomize: true,
  deleted: false,
};

describe('ExerciseView', () => {
  let model;
  beforeEach(() => {
    model = new Backbone.Model({
      title: 'test',
      changed: false,
      id: 'test',
      description: 'test',
      sort_order: 1.5,
      author: 'test',
      copyright_holder: 'test',
      license: 'test',
      language: 'test',
      license_description: 'test',
      assessment_items: [],
      files: [],
      parent_title: 'test',
      ancestors: [],
      modified: 'test',
      original_channel: 'test',
      kind: 'test',
      parent: 'test',
      children: 'test',
      published: 'test',
      associated_presets: [],
      valid: 'test',
      metadata: 'test',
      original_source_node_id: 'test',
      tags: 'test',
      extra_fields: {
        randomize: false,
      },
      prerequisite: [],
      is_prerequisite_of: 'test',
      node_id: 'test',
      tree_id: 'test',
      publishing: false,
      freeze_authoring_data: false,
      role_visibility: 'test',
      provider: 'test',
      aggregator: 'test',
      thumbnail_src: 'test',
    });
  });
  describe('render method', () => {
    it('should render with a valid content node with no assessment_items', () => {
      const view = new ExerciseView({
        model,
      });
      view.render();
      const html = view.$el[0].outerHTML;
      expect(html).toMatchSnapshot();
    });
    it('should render with a valid content node with 1 assessment_items', () => {
      model.set('assessment_items', [exampleAssessment]);
      const view = new ExerciseView({
        model,
      });
      view.render();
      const html = view.$el[0].outerHTML;
      expect(html).toMatchSnapshot();
    });
    it('should render with a valid content node with 3 different assessment_items', () => {
      const exampleAssessment1 = exampleAssessment;
      const exampleAssessment2 = Object.assign({}, exampleAssessment);
      exampleAssessment2.type = 'single_selection';
      const exampleAssessment3 = Object.assign({}, exampleAssessment);
      exampleAssessment3.type = 'single_selection';
      exampleAssessment3.answers = '[]';
      model.set('assessment_items', [exampleAssessment1, exampleAssessment2, exampleAssessment3]);
      const view = new ExerciseView({
        model,
      });
      view.render();
      const html = view.$el[0].outerHTML;
      expect(html).toMatchSnapshot();
    });
  });
});
