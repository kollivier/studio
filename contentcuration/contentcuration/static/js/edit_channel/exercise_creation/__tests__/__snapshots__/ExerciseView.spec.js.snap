// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseView render method should render with a valid content node with 1 assessment_items 1`] = `
<div>
  <div class="container-fluid" id="exercisecreation">
    <div class="container-fluid exercise_header_bar">
      <div class="toggle_question_order pull-right">
        <input id="randomize_question_order" class="randomize_exercise" type="checkbox" disabled="">
        <label for="randomize_question_order" class="randomize_exercise"> Randomize question order for learners</label>
      </div>
      <div class="show_question_toggle">
        <input type="checkbox" id="exercise_show_answers">
        <label for="exercise_show_answers"> View questions only</label>
      </div>
    </div>
    <br>
    <ol class="panel-group" id="exercise_list" role="tablist" aria-multiselectable="true">
      <div class="default-item container-fluid text-center" style="display: none;">No questions associated with this exercise</div>
      <li class="assessment_li">
        <div class="input-tab-control tab_item" data-next="#mastery_modal_close_button"></div>
        <div class="assessment_order_area text-center pull-left">
          <div class="order"></div>
        </div>
        <div class="assessment_item container-fluid ">
          <div class="main_edit_area">
            <div class="question_area toggle_exercise">
              <div class="row question_controls">
                <div class="col-sm-5">
                  <div class="question_type">Unknown Question Type</div>
                  <select class="question_type_select show_on_focus" value="multiplechoice">
                    <option class="single_selection" value="single_selection">Single Selection</option>
                    <option class="multiple_selection" value="multiple_selection">Multiple Selection</option>
                    <option class="true_false" value="true_false">True/False</option>
                    <option class="input_question" value="input_question">Input Question</option>
                  </select>
                </div>
                <div class="col-sm-3 hint_link_cell">
                  <div class="open_hints hint_link pull-right uppercase">
                    <span class="hint_count">1 Hint</span>&nbsp;
                    <i class="material-icons">launch</i>
                  </div>
                </div>
                <div class="col-sm-4 random_answers_order">
                  <div class="random_answer_section">
                    <input type="checkbox" id="random_answer_order_view38" checked="" disabled="" class="random_order_check">
                  </div>
                  <div class="random_answer_section">
                    <label for="random_answer_order_view38" disabled="">Randomize Answer Order</label>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <ul class="error-list list-unstyled text-center"></ul>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <div class="question editable" data-ph="Question">
                    <div id="editor_view_view39"></div>
                  </div>
                </div>
              </div>
            </div>
            <div id="" class="list_area">
              <div class="answers">
                <div>
                  <div class="show_on_focus answer_header">
                    <div class="col-sm-1 correct_header">
                      <i class="material-icons align-bottom" title="Correct?">check_circle</i>
                    </div>
                    <div class="col-sm-11 uppercase">Answers</div>
                  </div>

                  <ul class="list-unstyled answer_list">
                    <div class="default-item" style="display: none;">No answers provided.</div>
                    <li class="answer_li">
                      <div class="answer_item  is_correct">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c8" disabled="" checked="" title="Correct">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view44"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c8" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view48"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c8" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view52"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                  </ul>
                  <div class="show_on_focus addanswer" title="Add">
                    <i class="material-icons align-bottom">add</i>&nbsp; <span class="addanswertext uppercase">Answer</span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="remove_area">
            <div class="toolbar"></div>
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`ExerciseView render method should render with a valid content node with 3 different assessment_items 1`] = `
<div>
  <div class="container-fluid" id="exercisecreation">
    <div class="container-fluid exercise_header_bar">
      <div class="toggle_question_order pull-right">
        <input id="randomize_question_order" class="randomize_exercise" type="checkbox" disabled="">
        <label for="randomize_question_order" class="randomize_exercise"> Randomize question order for learners</label>
      </div>
      <div class="show_question_toggle">
        <input type="checkbox" id="exercise_show_answers">
        <label for="exercise_show_answers"> View questions only</label>
      </div>
    </div>
    <br>
    <ol class="panel-group" id="exercise_list" role="tablist" aria-multiselectable="true">
      <div class="default-item container-fluid text-center" style="display: none;">No questions associated with this exercise</div>
      <li class="assessment_li">
        <div class="input-tab-control tab_item" data-next="#mastery_modal_close_button"></div>
        <div class="assessment_order_area text-center pull-left">
          <div class="order"></div>
        </div>
        <div class="assessment_item container-fluid ">
          <div class="main_edit_area">
            <div class="question_area toggle_exercise">
              <div class="row question_controls">
                <div class="col-sm-5">
                  <div class="question_type">Unknown Question Type</div>
                  <select class="question_type_select show_on_focus" value="multiplechoice">
                    <option class="single_selection" value="single_selection">Single Selection</option>
                    <option class="multiple_selection" value="multiple_selection">Multiple Selection</option>
                    <option class="true_false" value="true_false">True/False</option>
                    <option class="input_question" value="input_question">Input Question</option>
                  </select>
                </div>
                <div class="col-sm-3 hint_link_cell">
                  <div class="open_hints hint_link pull-right uppercase">
                    <span class="hint_count">1 Hint</span>&nbsp;
                    <i class="material-icons">launch</i>
                  </div>
                </div>
                <div class="col-sm-4 random_answers_order">
                  <div class="random_answer_section">
                    <input type="checkbox" id="random_answer_order_view126" checked="" disabled="" class="random_order_check">
                  </div>
                  <div class="random_answer_section">
                    <label for="random_answer_order_view126" disabled="">Randomize Answer Order</label>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <ul class="error-list list-unstyled text-center"></ul>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <div class="question editable" data-ph="Question">
                    <div id="editor_view_view127"></div>
                  </div>
                </div>
              </div>
            </div>
            <div id="" class="list_area">
              <div class="answers">
                <div>
                  <div class="show_on_focus answer_header">
                    <div class="col-sm-1 correct_header">
                      <i class="material-icons align-bottom" title="Correct?">check_circle</i>
                    </div>
                    <div class="col-sm-11 uppercase">Answers</div>
                  </div>

                  <ul class="list-unstyled answer_list">
                    <div class="default-item" style="display: none;">No answers provided.</div>
                    <li class="answer_li">
                      <div class="answer_item  is_correct">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c58" disabled="" checked="" title="Correct">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view132"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c58" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view136"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="checkbox" aria-label="Correct" name="answer_c58" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view140"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                  </ul>
                  <div class="show_on_focus addanswer" title="Add">
                    <i class="material-icons align-bottom">add</i>&nbsp; <span class="addanswertext uppercase">Answer</span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="remove_area">
            <div class="toolbar"></div>
          </div>
        </div>
      </li>
      <li class="assessment_li">
        <div class="input-tab-control tab_item" data-next="#mastery_modal_close_button"></div>
        <div class="assessment_order_area text-center pull-left">
          <div class="order"></div>
        </div>
        <div class="assessment_item container-fluid ">
          <div class="main_edit_area">
            <div class="question_area toggle_exercise">
              <div class="row question_controls">
                <div class="col-sm-5">
                  <div class="question_type">Single Selection</div>
                  <select class="question_type_select show_on_focus" value="single_selection">
                    <option class="single_selection" value="single_selection">Single Selection</option>
                    <option class="multiple_selection" value="multiple_selection">Multiple Selection</option>
                    <option class="true_false" value="true_false">True/False</option>
                    <option class="input_question" value="input_question">Input Question</option>
                  </select>
                </div>
                <div class="col-sm-3 hint_link_cell">
                  <div class="open_hints hint_link pull-right uppercase">
                    <span class="hint_count">1 Hint</span>&nbsp;
                    <i class="material-icons">launch</i>
                  </div>
                </div>
                <div class="col-sm-4 random_answers_order">
                  <div class="random_answer_section">
                    <input type="checkbox" id="random_answer_order_view144" checked="" disabled="" class="random_order_check">
                  </div>
                  <div class="random_answer_section">
                    <label for="random_answer_order_view144" disabled="">Randomize Answer Order</label>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <ul class="error-list list-unstyled text-center"></ul>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <div class="question editable" data-ph="Question">
                    <div id="editor_view_view145"></div>
                  </div>
                </div>
              </div>
            </div>
            <div id="" class="list_area">
              <div class="answers">
                <div>
                  <div class="show_on_focus answer_header">
                    <div class="col-sm-1 correct_header">
                      <i class="material-icons align-bottom" title="Correct?">check_circle</i>
                    </div>
                    <div class="col-sm-11 uppercase">Answers</div>
                  </div>

                  <ul class="list-unstyled answer_list">
                    <div class="default-item" style="display: none;">No answers provided.</div>
                    <li class="answer_li">
                      <div class="answer_item  is_correct">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="radio" aria-label="Correct" name="answer_c63" disabled="" checked="" title="Correct">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view150"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="radio" aria-label="Correct" name="answer_c63" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view154"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                    <li class="answer_li">
                      <div class="answer_item  ">
                        <div class="row">
                          <div class="col-xs-1 text-center">
                            <input class="correct" type="radio" aria-label="Correct" name="answer_c63" disabled="" title="Incorrect">
                          </div>
                          <div class="col-xs-8 editor-col">
                            <div class="answer editable" data-ph="Answer">
                              <div id="editor_view_view158"></div>
                            </div>
                          </div>
                          <div class="col-xs-3 answer_options">

                          </div>
                        </div>
                      </div>
                    </li>
                  </ul>
                  <div class="show_on_focus addanswer" title="Add">
                    <i class="material-icons align-bottom">add</i>&nbsp; <span class="addanswertext uppercase">Answer</span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="remove_area">
            <div class="toolbar"></div>
          </div>
        </div>
      </li>
      <li class="assessment_li">
        <div class="input-tab-control tab_item" data-next="#mastery_modal_close_button"></div>
        <div class="assessment_order_area text-center pull-left">
          <div class="order"></div>
        </div>
        <div class="assessment_item container-fluid ">
          <div class="main_edit_area">
            <div class="question_area toggle_exercise">
              <div class="row question_controls">
                <div class="col-sm-5">
                  <div class="question_type">Single Selection</div>
                  <select class="question_type_select show_on_focus" value="single_selection">
                    <option class="single_selection" value="single_selection">Single Selection</option>
                    <option class="multiple_selection" value="multiple_selection">Multiple Selection</option>
                    <option class="true_false" value="true_false">True/False</option>
                    <option class="input_question" value="input_question">Input Question</option>
                  </select>
                </div>
                <div class="col-sm-3 hint_link_cell">
                  <div class="open_hints hint_link pull-right uppercase">
                    <span class="hint_count">1 Hint</span>&nbsp;
                    <i class="material-icons">launch</i>
                  </div>
                </div>
                <div class="col-sm-4 random_answers_order">
                  <div class="random_answer_section">
                    <input type="checkbox" id="random_answer_order_view162" checked="" disabled="" class="random_order_check">
                  </div>
                  <div class="random_answer_section">
                    <label for="random_answer_order_view162" disabled="">Randomize Answer Order</label>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <ul class="error-list list-unstyled text-center"></ul>
                </div>
              </div>
              <div class="row">
                <div class="col-sm-12">
                  <div class="question editable" data-ph="Question">
                    <div id="editor_view_view163"></div>
                  </div>
                </div>
              </div>
            </div>
            <div id="" class="list_area">
              <div class="answers">
                <div>
                  <div class="show_on_focus answer_header">
                    <div class="col-sm-1 correct_header">
                      <i class="material-icons align-bottom" title="Correct?">check_circle</i>
                    </div>
                    <div class="col-sm-11 uppercase">Answers</div>
                  </div>

                  <ul class="list-unstyled answer_list">
                    <div class="default-item" style="display: block;">No answers provided.</div>
                  </ul>
                  <div class="show_on_focus addanswer" title="Add">
                    <i class="material-icons align-bottom">add</i>&nbsp; <span class="addanswertext uppercase">Answer</span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="remove_area">
            <div class="toolbar"></div>
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`ExerciseView render method should render with a valid content node with no assessment_items 1`] = `
<div>
  <div class="container-fluid" id="exercisecreation">
    <div class="container-fluid exercise_header_bar">
      <div class="toggle_question_order pull-right">
        <input id="randomize_question_order" class="randomize_exercise" type="checkbox" disabled="">
        <label for="randomize_question_order" class="randomize_exercise"> Randomize question order for learners</label>
      </div>
      <div class="show_question_toggle">
        <input type="checkbox" id="exercise_show_answers">
        <label for="exercise_show_answers"> View questions only</label>
      </div>
    </div>
    <br>
    <ol class="panel-group" id="exercise_list" role="tablist" aria-multiselectable="true">
      <div class="default-item container-fluid text-center" style="display: block;">No questions associated with this exercise</div>
    </ol>
  </div>
</div>
`;
